### PR TITLE
Update trust-policies chart with support for trust domain

### PR DESCRIPTION
**Pull request by codysoyland**
Fixes https://github.com/github/package-security/issues/2105

Signed-off-by: Cody Soyland <codysoyland@github.com>

### CODE REVIEW

**Review by ejahnGithub — APPROVED** [review]
lgtm

File: charts/trust-policies/values.yaml [hunk 1]
```diff
@@ -25,5 +25,10 @@ policy:
   trust:
     # trust the GitHub signing authority
     github: true
+    # githubTrustDomain may be set to the trust domain applicable for some GitHub Enterprise Cloud accounts.
+    # If applicable, this may be set to the output of the following command:
+    # $ gh api meta | jq .domains.artifact_attestations.trust_domain
+    # See documentation for more details: https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/enforcing-artifact-attestations-with-a-kubernetes-admission-controller
```
Comment (steiza) [inline charts/trust-policies/values.yaml:31]: Minor: I don't see any mention of trust domains on this documentation today - maybe it's getting updated later?

File: charts/trust-policies/Chart.yaml [hunk 2]
```diff
@@ -8,8 +8,8 @@ sources:
 type: application
 
 name: trust-policies
-version: "v0.5.0"
-appVersion: "v0.5.0"
+version: "v0.6.0"
```
Comment (steiza) [inline charts/trust-policies/Chart.yaml:11]: Minor / just to make sure I understand: we're making this tag after this pull request lands?

**Review by steiza — APPROVED** [review]
Minor non-blocking Q's, LGTM!

File: charts/trust-policies/values.yaml (hunk 1 above)
Comment (phillmv) [reply charts/trust-policies/values.yaml:31]: @steiza separate PR

File: charts/trust-policies/values.yaml [hunk 3]
```diff
@@ -25,5 +25,10 @@ policy:
   trust:
     # trust the GitHub signing authority
     github: true
+    # githubTrustDomain may be set to the trust domain applicable for some GitHub Enterprise Cloud accounts.
+    # If applicable, this may be set to the output of the following command:
+    # $ gh api meta | jq .domains.artifact_attestations.trust_domain
```
Comment (phillmv) [inline charts/trust-policies/values.yaml:30]:
```suggestion
    # $ gh api meta -q .domains.artifact_attestations.trust_domain
```

Comment (phillmv) [reply charts/trust-policies/values.yaml:30]: (turns out gh ships with jq?)

File: charts/trust-policies/Chart.yaml (hunk 2 above)
Comment (codysoyland) [reply charts/trust-policies/Chart.yaml:11]: That's right!

File: charts/trust-policies/values.yaml (hunk 3 above)
Comment (codysoyland) [reply charts/trust-policies/values.yaml]: Good call! I pushed a new commit using the long form flag `--jq`.

**Review by steiza — APPROVED** [review]
The setting that keeps dismissing reviews is quite annoying...

**Pull request merged**